### ISSUE TITLE
virt_sandbox_use_nfs

### ISSUE THREAD

**Issue by edseymour** [fedora-cloud/docker-selinux#18]
I was under the impression that docker-selinux added the `virt_sandbox_use_nfs` boolean, however, I'm not getting added, only `docker_connect_any`. this is on Fedora 23 desktop and replicated on a Fedora 24 server build.

**Comment by rhatdan** [comment]
virt_sandbox_use_nfs is defined in virt.te notice the virt_ prefix as opposed to the docker_ prefix.  We temporarily shipped it in docker-selinux until selinux-policy could be updated though.  Hence the confusion.

**Comment by edseymour** [comment]
I thought it might have been something like that, thank you for clearing it up!

**Comment by edseymour** [comment]
So I think the rpm for docker-selinux has the virt_ policies removed, but they don't appear to be moved into the selinux-policy, AFAIK. 

F24 server, selinux policy installed rpms
```
selinux-policy-3.13.1-191.fc24.3.noarch
selinux-policy-targeted-3.13.1-191.fc24.3.noarch
docker-selinux-1.10.3-21.git19b5791.fc24.x86_64
selinux-policy-sandbox-3.13.1-191.fc24.3.noarch
```
and 
```
[root@myserver ~]# getsebool -a | grep sandbox
unconfined_chrome_sandbox_transition --> on
virt_sandbox_use_all_caps --> on
virt_sandbox_use_audit --> on
virt_sandbox_use_fusefs --> off
virt_sandbox_use_mknod --> off
virt_sandbox_use_netlink --> off
virt_sandbox_use_sys_admin --> off
```

**Comment by rhatdan** [comment]
Looks like it is removed from selinux-policy upstream.  You can just turn on virt_use_nfs to turn it on.  I will talk to the selinux guys about this.

**Comment by edseymour** [comment]
Thanks @rhatdan I will do that. It was breaking the openshift origin install as the playbook was checking for this; I wanted to know if I should hack the playbook (I guess, yes) or configure something.

**Comment by rhatdan** [comment]
@wrabcak We need the following patch added to selinux-policy package for F24, Rawhide and RHEL7.3

```
diff --git a/booleans.subs_dist b/booleans.subs_dist
index d4ff62c..fed7d8c 100644
--- a/booleans.subs_dist
+++ b/booleans.subs_dist
@@ -51,3 +51,4 @@ clamd_use_jit antivirus_use_jit
 amavis_use_jit antivirus_use_jit
 logwatch_can_sendmail logwatch_can_network_connect_mail
 puppet_manage_all_files puppetagent_manage_all_files
+virt_sandbox_use_nfs virt_use_nfs
```

**Comment by rhatdan** [comment]
You can modify this file on your host and the install should work

/etc/selinux/targeted/booleans.subs_dist

**Comment by rhatdan** [comment]
Yes I think we have dropped this boolean and are just going to use virt_use_nfs.

**Comment by edseymour** [comment]
@rhatdan moving to virt_use_nfs may be the plan... I'm not sure it's been implemented yet. I get the feeling we're currently in a halfway house.

virt_use_nfs is on, I noticed a (new?) boolean openshift_use_nfs, which was off, so I switched this to on. However, it looks like SELINUX policies are denying a container from access an NFS mountpoint. 

The example I'm using is the Origin docker-registry, which in my configuration is mounting /registry within the container. The mount point cannot be accessed, for example, `ls /registry/` from within the container returns `permission denied`. 

On the host the `/var/log/audit/audit.log` reports
```
type=AVC msg=audit(1468830712.903:35320): avc:  denied  { read } for  pid=15859 comm="ls" name="registry" dev="0:193" ino=1158766719 scontext=system_u:system_r:svirt_lxc_net_t:s0:c0,c1 tcontext=unconfined_u:object_r:unlabeled_t:s0 tclass=dir permissive=0
```

**Comment by rhatdan** [comment]
This is n ot showing /registry as a nfs share it is showing as unlabeled_t, which means there is no labels on this directory.  Are you volume mounting a directory into the container?

**Comment by edseymour** [comment]
@rhatdan yeah, odd right? Using Kubernetes PersistentVolume with an associated PVC for the registry. The PV references an NFS server. I'll re-run and confirm

**Comment by rhatdan** [comment]
Could you run restorecon on /registry?

**Comment by edseymour** [comment]
@rhatdan (sorry for taking a little while to get back).

the /registry directory is *inside* the container, running within OpenShift and actually mounted from an NFS export from another (Fedora) system, so I don't think I can run restorecon there. 

What I could see, however, was that the SELinux labels were being pulled through from the labels on the NFS server. It was some arbitrary directory being exported, so restorecon wouldn't help much, so instead I relabelled the directories, on the NFS server, with the `svirt_sandbox_t` label. 

`# chcon -Rt svirt_sandbox_file_t /path/to/my/export`

The label was then pulled through when running the container, _and_ I was now able to write to the directory. (So I'm happy with that!)

Thanks

**Comment by rhatdan** [comment]
If /registry is from an NFS share it should be labeled nfs_t?

**Comment by edseymour** [comment]
@rhatdan yes, you're right - I was thinking of local host container mounts. Adding the `nfs_t` label works.
